humatest/humatest.go

```diff
@@ -81,9 +91,12 @@ type TestAPI interface {
 	//
 	// 	// Make a PUT request with a custom header.
 	// 	api.Put("/foo", "X-My-Header: my-value", MyBody{Foo: "bar"})
+	PutCtx(ctx context.Context, path string, args ...any) *httptest.ResponseRecorder
+
+	// Put wraps [TestAPI.PutCtx] using [context.Background].
```

Comment:
For this one and others I think you should address these:

- the comment should be taken from existing code, here your changes lead to have less information in the comment, so in the godoc
- you should add something to existing comment to mention PutCtx should be preferred when there is an existing context

Reply:
Ah, I didn't realize comments inside interfaces don't support links in pkgsite (it does in most editors, for example). I've copied the comment over and edited them to reflect the correct signature.

Reply:
I think you missed or ignored my second comment.

The fact you could use in each old code there is a version with a context that should be preferred

Reply:
Ah, sorry it seemed I missed or ignored this. Let me clarify!

The comments do make some case for the scenarios you'd use it, perhaps implicitly
> Get performs a GET request against the API using [context.Background] in the
	// [http.Request]
	
While it's true this is an implicit instruction, I'd expect anyone familiar with Go to understand when they do or do not need to provide their own context.

To address preference, I don't necessarily think from the perspective of the library there is a preference - this is simply a utility after all. Similarly, `http` has no preference if users use `NewRequest` or `NewRequestWithContext`, it's just an option. I'd expect anyone using this library to be able to make their own choices about whether or not they need to provide a context. That said, I didn't update the website documentation (which in practice, is where I get most of the examples from), so please look at https://github.com/danielgtaylor/huma/pull/469/commits/da24dacc085ae5188f67114a6345dd85d65db609.